src/routes/auth/user.ts

```diff
@@ -23,77 +23,89 @@ export interface UserLoginData {
 
 //user register route
 userAuthRouter.post("/register", async (req, res) => {
-  const newUser: UserRegisterData = { email: req.body.email, username: req.body.username, password: req.body.password };
-
-  const { valid, errors } = validateUserSignupData(newUser);
-
-  if (!valid) return res.status(400).json(errors);
-
-  //checking if user doesn't already exist in DB
-  const userUsernameDoc = await UserModel.findOne({ username: newUser.username }).exec();
-  const userEmailDoc = await UserModel.findOne({ email: newUser.email }).exec();
-  if (userUsernameDoc) return res.status(400).json({ username: "this username is already taken" });
-  if (userEmailDoc) return res.status(400).json({ email: "this email is already taken" });
-
-  // Gen salt and store it together with the hashed password
-  const userPasswordSalt = await bcrypt.genSalt(10);
-  const hashedPassword = <string>await bcrypt.hash(newUser.password, userPasswordSalt).catch((err) => console.log(err));
-
-  // Add user to DB
-  const _newUserModel = (
-    await UserModel.create({
-      email: newUser.email,
-      username: newUser.username,
-      password: hashedPassword,
-      password_salt: userPasswordSalt,
-      private_profile: false,
-      banned: false,
-    })
-  ).save();
-
-  // Return auth token;
-  let accessToken;
-  const TOKEN = process.env.ACCESS_TOKEN_SECRET;
-  TOKEN ? (accessToken = jwt.sign({ email: newUser.email, username: newUser.username }, TOKEN)) : console.log("No secret access token in env");
-
-  res.status(200).json({ jwt: `Bearer ${accessToken}` });
+  try {
+    const newUser: UserRegisterData = { email: req.body.email, username: req.body.username, password: req.body.password };
+
+    const { valid, errors } = validateUserSignupData(newUser);
+
+    if (!valid) return res.status(400).json(errors);
+
+    //checking if user doesn't already exist in DB
+    const userUsernameDoc = await UserModel.findOne({ username: newUser.username }).exec();
+    const userEmailDoc = await UserModel.findOne({ email: newUser.email }).exec();
+    if (userUsernameDoc) return res.status(400).json({ username: "this username is already taken" });
+    if (userEmailDoc) return res.status(400).json({ email: "this email is already taken" });
+
+    // Gen salt and store it together with the hashed password
+    const userPasswordSalt = await bcrypt.genSalt(10);
+    const hashedPassword = <string>await bcrypt.hash(newUser.password, userPasswordSalt);
```

Comment:
You don't need to cast to string since you're using await and the result will always be a string (otherwise it will throw and there will be no result at all).